qa/rpc-tests/importmulti.py

```diff
@@ -0,0 +1,157 @@
+        address_info = self.nodes[1].validateaddress(address1)
+        assert_equal(address_info['iswatchonly'], False)
+        assert_equal(address_info['ismine'], False)
+
+        address_info = self.nodes[1].validateaddress(address2)
+        assert_equal(address_info['iswatchonly'], False)
+        assert_equal(address_info['ismine'], False)
+
+        address_info = self.nodes[1].validateaddress(address3)
+        assert_equal(address_info['iswatchonly'], False)
+        assert_equal(address_info['ismine'], False)
+
+        # import multi
+        result1 = self.nodes[1].importmulti( [
+            { "type": ADDRESS_KEY, "value": address1 , "label":"new account 1" , "timestamp": begintime } ,
+            { "type": PUB_KEY , "value": address2_pubkey , "label":"new account 1", "timestamp": begintime},
+            { "type": PRIV_KEY , "value": address3_privkey , "timestamp": begintime},
+            { "type": SCRIPT_KEY , "value": address4_scriptpubkey , "timestamp": begintime},
+            ])
+
+        #Addresses Test - after import
+        address_info = self.nodes[1].validateaddress(address1)
+        assert_equal(address_info['iswatchonly'], True)
+        assert_equal(address_info['ismine'], False)
+        address_info = self.nodes[1].validateaddress(address2)
+        assert_equal(address_info['iswatchonly'], True)
+        assert_equal(address_info['ismine'], False)
+        address_info = self.nodes[1].validateaddress(address3)
+        assert_equal(address_info['iswatchonly'], False)
+        assert_equal(address_info['ismine'], True)
+        address_info = self.nodes[1].validateaddress(address4)
+        assert_equal(address_info['iswatchonly'], True)
+        assert_equal(address_info['ismine'], False)
+
+        assert_equal(result1[0]['success'], True)
+        assert_equal(result1[1]['success'], True)
+        assert_equal(result1[2]['success'], True)
+        assert_equal(result1[3]['success'], True)
+
+        #importmulti without rescan
+        result2 = self.nodes[1].importmulti( [
+            { "type": ADDRESS_KEY, "value": self.nodes[0].getnewaddress() } ,
+            { "type": ADDRESS_KEY, "value": self.nodes[0].getnewaddress() } ,
+            { "type": ADDRESS_KEY, "value": self.nodes[0].getnewaddress() , "label":"random account" } ,
+            { "type": PUB_KEY, "value": self.nodes[0].validateaddress(self.nodes[0].getnewaddress())['pubkey'] } ,
+            { "type": SCRIPT_KEY, "value": self.nodes[0].validateaddress(self.nodes[0].getnewaddress())['scriptPubKey'] },
+            ], { "rescan":False } )
+
+        # all succeed
+        assert_equal(result2[0]['success'], True)
+        assert_equal(result2[1]['success'], True)
+        assert_equal(result2[2]['success'], True)
+        assert_equal(result2[3]['success'], True)
+        assert_equal(result2[4]['success'], True)
+
+        # empty json case
+        try:
+            self.nodes[1].importmulti()
+            raise
```

Comment:
Nit: Would be nice to be more verbose here. Maybe move the comment from above into an `AssertionError`?

```py
raise AssertionError("must not accept empty json")